builtin/providers/test/resource_with_custom_diff.go

```diff
@@ -0,0 +1,84 @@
+package test
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func testResourceCustomDiff() *schema.Resource {
+	return &schema.Resource{
+		Create:        testResourceCustomDiffCreate,
+		Read:          testResourceCustomDiffRead,
+		CustomizeDiff: testResourceCustomDiffCustomizeDiff,
```

Comment:
Ahh yes, of course. I forgot that there are actually _methods_ on this struct. :grinning:

I think part of why this is hard to name is that this function kinda serves two purposes:

* Check whether a diff is valid
* Alter a diff to provide more context

So `AlterDiff` seems promising if we consider the checking to be an implicit side-effect, but that's still longer than all of the other verbs here, so doesn't really make a great deal of difference.

So with all of that said, maybe let's just stick with `CustomizeDiff` and accept the weird visual texture it creates... too minor a detail to have sweated this much over it. :grinning: